### ISSUE TITLE
Allow a count for rollback/migrate.

### ISSUE THREAD

**Issue by rquadling** [cakephp/phinx#258]
Hi.

A feature request.

I'm doing some incremental development, but I need to be able to test things backwards and forwards through the migrations.

Currently, I have to track which timestamp to use as my target, but I can see it is always 3 migrations (or 4, 5, etc. as I develop).

A new option of `--count (-o) Number of migrations to process` would be useful for this.

Regards,

Richard Quadling.

**Comment by ipoz** [comment]
:+1:

**Comment by rquadling** [comment]
Now I'm more familiar with Phinx, I'll create a feature branch for a PR later this (or early next) week.

**Comment by steffkes** [comment]
Having used MyBatis Migrations before, they're doing the following:

> The **up** command runs the **do** section of all pending migrations in order, one after the other.
> The **down** command runs the **undo** section of the *last applied migration* only.
http://mybatis.github.io/migrations/updown.html

while specifying the number of migrations to run is obviously fine as well, i've seen this behavior used by multiple migration libraries .. so we should probably go for that?

**Comment by rquadling** [comment]
I'm no longer sure this is a useful feature for me as I developed the breakpoint feature https://github.com/rquadling/phinx/tree/feature/BreakpointSupport / https://github.com/robmorgan/phinx/pull/497 which is what I really needed.

Waiting on some feedback from Phinx management on this.

**Comment by robmorgan** [comment]
I see this as pretty low priority.

**Comment by dereuromark** [comment]
The current timestamps without autocomplete are sure not nice to work with.
A different way, like this count sure could be more user friendly.

**Comment by othercorey** [comment]
Closing due to lack of activity or out of scope. If there is new interest, please open a PR.

**Comment by MasterOdin** [comment]
@othercorey could this be re-opened?

**Comment by dereuromark** [comment]
@MasterOdin Do we need to keep this open?
It is useful, but it seems there is no interest in adding it. Once a PR here is done in the future, that would still be fine.
But tickets like these seem to be just dead ones.

**Comment by MasterOdin** [comment]
Count command has been added to migrate with `-k` / `--count` option flag. Need to add it to rollback still.

**Comment by dereuromark** [comment]
https://github.com/cakephp/migrations/pull/874
can be ported back on the rollback part - then this ticket would be fully resolved - 11 years after the fact^^
